src/main/resources/static/surveyor.js

```diff
@@ -39,10 +39,210 @@ $(document).ready(function() {
+
+            switch (classname) {
+                //open ended questions
+                case 'openEndedQuestion':
+
+                    //Create JSON Object for openEndedQuestion
+                    jsonQuestions.push(JSON.stringify({
+                        "question": questionText,
+                    }));
+                    break
+
+                //Numerical range questions
+                case 'numberRangeQuestion':
+                    let upperBound = parseInt($(this).find("#UpperBound").val());
+                    let lowerBound = parseInt($(this).find("#LowerBound").val());
+
+                    //check if its not a number
+                    if(isNaN(upperBound) || isNaN(lowerBound)) {
+                        alert("Upper Bound or Lower Bound is not a value for question " + questionText)
+                        error = true;
+                        return
+                    }
+
+                    //Check if the bounds are empty
+                    if (!upperBound || !lowerBound) {
+                        alert("Must fill in both upper bound and lower bound for question: " + questionText)
+                        error = true;
+                        return
+                    }
+
+                    //upperBound must always be greater than lowerBound
+                    else if (upperBound < lowerBound) {
+                        alert("Upper Bound must be a larger value than Lower Bound for question: " + questionText)
+                        error = true;
+                        return
+                    }
+                    else{
+                    //Create JSON Object for Numerical Range Question
+                        jsonQuestions.push(JSON.stringify({
+                            "question": questionText,
+                            "lowerBound": lowerBound,
+                            "upperBound": upperBound
+                        }));
+                    }
+                    // console.log(numericalRangeQuestions)
+                    break
+
+                //Multiple choice questions
+                case 'multipleChoiceQuestion':
+
+                    //Declare optionsJSON
+                    let optionsJSON = {}
+
+                    //Keep track of current option number
+                    let count = 1;
+
+                    //loop through each mcq option
+                    $(this).find(".MCQOption").each(function () {
+                        let option = $(this).val();
+                        console.log(option);
+
+                        //Check if all the MCQ options are filled
+                        if (!option) {
+                            alert("One of the MCQ options is empty for question: " + questionText);
+                            error = true
+                            return
+                        }
+
+                        //Add to data structure
+                        optionsJSON[option] = count
```

Comment:
we dont need count. just need to set all int to 0. e.g.
{
  "options": {
    "first option": 0,
    "second option": 0,
    "third opton": 0
  },
  "question": "Question String"
}